### PR TITLE
Use poppler-utils from stack

### LINKED ISSUE #21: Invalidate the cache on stack upgrade
Hi

I noticed that currently the buildpack doesn't invalidate the cache when the app's `$STACK` changes, which will cause issues depending on changes to the format of APT's cache between Ubuntu versions. (And even if APT is safely migrates/ignores old files, the cache would be bloated with the previous distro's .deb files, so it would be good to clean it out regardless.)

A possible pattern for solving this would be:
https://github.com/heroku/heroku-buildpack-apt/blob/0357ec1e8863a803ada6c2d57f0850f2c5fa0366/bin/compile#L32-L41

Many thanks :-)

### PR DESCRIPTION
**Pull request by dzuelke**
Current `main` and `update-ffmpeg` will now fail, as the recent stack image update brought in all the libraries that `poppler-utils` needs as part of the `libvips` install.

As a result, the `find` for `.so` files piped to `xargs dirname` in `bin/compile` will fail, because no `.so` files are actually added:

```
remote: -----> ActiveStorage Preview app detected
remote: -----> Installing binary dependencies for ActiveStorage Preview
remote:        Detected ffmpeg version change, flushing cache
remote:        Downloading packagesW: http://apt.postgresql.org/pub/repos/apt/dists/jammy-pgdg/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
remote: ..
remote:        Installing packages..
remote: dirname: missing operand
remote: Try 'dirname --help' for more information.
remote:  !     Push rejected, failed to compile ActiveStorage Preview app.
remote: 
remote:  !     Push failed
```

This simplifies `bin/compile` greatly by removing all the now-unnecessary Apt trickery:

```
% heroku buildpacks:set -i 1 https://github.com/heroku/heroku-buildpack-activestorage-preview.git#stack-poppler
Buildpack set. Next release on blooming-castle-09486 will use:
  1. https://github.com/heroku/heroku-buildpack-activestorage-preview.git#stack-poppler
  2. heroku/ruby
Run git push heroku main to create a new release using these buildpacks.

% git push heroku master
…
remote: -----> ActiveStorage Preview app detected
remote: -----> Installing binary dependencies for ActiveStorage Preview
remote:        Downloading ffmpeg version 5.1.2
remote:        Installing ffmpeg
remote:        Done
remote:        Exporting environment variables
remote:        Done
…

% heroku run which ffmpeg pdftocairo
Running which ffmpeg pdftocairo on ⬢ blooming-castle-09486... up, run.3697 (Basic)
/app/.heroku/activestorage-preview/bin/ffmpeg
/usr/bin/pdftocairo
```

GUS-W-12618598

Also closes #21 

GUS-W-8276306

### CODE REVIEW

**Comment by dzuelke** [comment]
(CI will likely fail until rollout of https://devcenter.heroku.com/changelog-items/2549 has propagated)

**Comment by dzuelke** [comment]
Rebased on top of `update-ffmpeg`.

Once ready, **please merge this after #31, and using a regular merge commit to ease merging/rebasing of stacked PR #34**

File: bin/compile
Comment (dzuelke) [reply bin/compile:42]: It still does, in line 16 :)

(the line you commented on actually uses `/tmp`, which is a more recent change added in #31)

**Review by edmorley — APPROVED** [review]

**Pull request merged**
